benchmarks/benchmarks/ag_methods.py

```diff
@@ -0,0 +1,321 @@
+from __future__ import division, absolute_import, print_function
+
+import MDAnalysis
+import numpy as np
+
+try:
+    from MDAnalysisTests.datafiles import GRO
+    from MDAnalysis.exceptions import NoDataError
+except:
+    pass
+
+class AtomGroupMethodsBench(object):
+    """Benchmarks for the various MDAnalysis
+    atomgroup methods.
+    """
+    # NOTE: the write() method has been
+    # excluded as file writing is considered
+    # a separate benchmarking category
+
+    params = (10, 100, 1000, 10000)
+    param_names = ['num_atoms']
+
+    def setup(self, num_atoms):
+        self.u = MDAnalysis.Universe(GRO)
+        self.ag = self.u.atoms[:num_atoms]
+        self.weights = np.ones(num_atoms)
+        self.vdwradii = {'NA':1.0,
+                         'M':1.0}
+        self.rot_matrix = np.ones((3,3))
+        self.trans = np.ones((4,4))
+
+    def time_bbox_pbc(self, num_atoms):
+        """Benchmark bounding box calculation
+        with pbc active.
+        """
+        self.ag.bbox(pbc=True)
+
+    def time_bbox_no_pbc(self, num_atoms):
+        """Benchmark bounding box calculation
+        with pbc inactive.
+        """
+        self.ag.bbox(pbc=False)
+
+    def time_bsphere_pbc(self, num_atoms):
+        """Benchmark bounding sphere calculation
+        with pbc active.
+        """
+        self.ag.bsphere(pbc=True)
+
+    def time_bsphere_no_pbc(self, num_atoms):
+        """Benchmark bounding sphere calculation
+        with pbc inactive.
+        """
+        self.ag.bsphere(pbc=False)
+
+    def time_center_pbc(self, num_atoms):
+        """Benchmark center calculation with
+        pbc active.
+        """
+        self.ag.center(weights=self.weights,
+                       pbc=True)
+
+    def time_center_no_pbc(self, num_atoms):
+        """Benchmark center calculation with
+        pbc inactive.
+        """
+        self.ag.center(weights=self.weights,
+                       pbc=False)
+
+    def time_centroid_pbc(self, num_atoms):
+        """Benchmark centroid calculation with
+        pbc active.
+        """
+        self.ag.centroid(pbc=True)
+
+    def time_centroid_no_pbc(self, num_atoms):
+        """Benchmark centroid calculation with
+        pbc inactive.
+        """
+        self.ag.centroid(pbc=False)
+
+    def time_concat(self, num_atoms):
```

Comment:
I'd suggest to use the same name as the method because in the graphs, it is labelled as `time_NAME` and it helps if `NAME` is identical to the method name `AtomGroup.NAME`. Adding option choices with underscores such as `_pbc` looks very sensible to me.